torch_xla/csrc/xla_lower_util.cpp

```diff
@@ -152,4 +152,34 @@ std::vector<xla::XlaOp> CreateBroadcastTensors(
   return result;
 }
 
+xla::XlaOp CreateIndex(const xla::XlaOp& input, const xla::XlaOp& indices) {
+  const xla::Shape input_shape = XlaHelpers::ShapeOfXlaOp(input);
+  const xla::Shape indices_shape = XlaHelpers::ShapeOfXlaOp(indices);
+  XLA_CHECK_GE(indices_shape.rank(), 1);
+  xla::int64 num_index_dims =
+      indices_shape.dimensions(indices_shape.rank() - 1);
+  xla::GatherDimensionNumbers dim_numbers;
+  std::vector<xla::int64> slice_sizes;
+  slice_sizes.reserve(input_shape.rank());
+  for (xla::int64 i = 0; i < input_shape.rank(); ++i) {
+    xla::int64 window_bound;
+    if (i < num_index_dims) {
+      dim_numbers.add_collapsed_slice_dims(i);
+      window_bound = 1;
+    } else {
+      window_bound = input_shape.dimensions(i);
+    }
+    slice_sizes.push_back(window_bound);
```

Comment:
I understand this comes from gather_op.cc, but I'd just call push_back() directly on the window_bound, instead of creating one temp only used in one place.
And we save two lines of code :)

Reply:
Done.